db-scripts/src/main/resources/cgds.sql

```diff
@@ -430,7 +430,7 @@ CREATE TABLE `mutation_event` (
   `KEYWORD` varchar(255) DEFAULT NULL COMMENT 'e.g. truncating, V200 Missense, E338del, ',
   KEY (`KEYWORD`),
   PRIMARY KEY (`MUTATION_EVENT_ID`),
-  UNIQUE (`CHR`, `START_POSITION`, `END_POSITION`, `TUMOR_SEQ_ALLELE`, `ENTREZ_GENE_ID`, `PROTEIN_CHANGE`, `MUTATION_TYPE`),
+  KEY `UQ_MUTATION_EVENT_DETAILS` (`CHR`, `START_POSITION`, `END_POSITION`, `TUMOR_SEQ_ALLELE`(255), `ENTREZ_GENE_ID`, `PROTEIN_CHANGE`, `MUTATION_TYPE`),
```

Comment:
@angelicaochoa @sheridancbio  is `KEY` a synonym to `INDEX`? If yes, please use `INDEX`, as `KEY` sounds a bit like this would be a unique or primary key (which I am assuming it is not...so please also remove the `UQ_` prefix of the index name).

Reply:
I think we have tried to match the format of CREATE TABLE statements that are generated by mysqldump. You can see other *plain* indexes annotated with KEY in the cgds_gdac table or by doing mysqldump --no-data. (For example, mutation_event has KEY `KEYWORD` (`KEYWORD`) and KEY `ENTREZ_GENE_ID` (`ENTREZ_GENE_ID`). To keep things consistent in the cgds.sql file, I think we should either use KEY or switch every plain index to INDEX. In mysql, the two words are synonymous I think.

Reply:
@pieterlukasse  I renamed the key but I agree with @sheridancbio - either we update everything or remain consistent with how the `cgds.sql` is currently set up

Reply:
They are generally synonymous, except that if it occurs in column definitions, `KEY` means `PRIMARY KEY` instead for compatibility reasons.

Source: https://dev.mysql.com/doc/refman/5.7/en/create-table.html#create-table-indexes-keys

Reply:
I see the difference you referenced @fedde-s ... I was not familiar with that. I've gone through cgds.sql looking for any usage of the keyword KEY in a column definition, and I didn't see any. For documentation, here are all the cases where a (non-column) KEY definition is present without being PRIMARY or UNIQUE or FOREIGN:

gene has KEY `HUGO_GENE_SYMBOL` (`HUGO_GENE_SYMBOL`)
uniprot_id_mapping (obsolete table) has KEY (`UNIPROT_ID`)
uniprot_id_mapping (obsolete table) has KEY (`UNIPROT_ACC`)
sample_profile has KEY (`SAMPLE_ID`)
mutation_event has KEY (`KEYWORD`)
mutation has KEY (`GENETIC_PROFILE_ID`,`ENTREZ_GENE_ID`)
mutation has KEY (`GENETIC_PROFILE_ID`,`SAMPLE_ID`)
mutation has KEY (`GENETIC_PROFILE_ID`)
mutation has KEY (`ENTREZ_GENE_ID`)
mutation has KEY (`SAMPLE_ID`)
mutation has KEY (`MUTATION_EVENT_ID`)
mutation_count_by_keyword has KEY (`GENETIC_PROFILE_ID`,`KEYWORD`)
drug has KEY `DRUG_NAME` (`DRUG_NAME`)
sample_cna_event has KEY (`GENETIC_PROFILE_ID`,`SAMPLE_ID`)
copy_number_seg has KEY (`CANCER_STUDY_ID`,`SAMPLE_ID`)
cosmic_mutation has KEY (`KEYWORD`)
clinical_event has KEY (`PATIENT_ID`, `EVENT_TYPE`)

If we want to refactor cgds.sql to use INDEX in place of KEY for these usages, I am not strongly opposed. I think it has been KEY because of the generated CREATE TABLE statements that result from mysqldump. It seems like a useful thing to be able to dump a database schema with mysqldump --no-data and compare the schema with cgds.sql to see how you differ from the "standard" cbioportal database schema. But perhaps skipping over these small differences by eye would not be a big deal since there are other transformations necessary as well (table ordering, removal of comments, etc)

Reply:
thanks @fedde-s , @sheridancbio , let's keep it as `KEY` for now :+1: